cmd/zpool/os/linux/zpool_vdev_os.c

```diff
@@ -88,7 +88,7 @@
 
 typedef struct vdev_disk_db_entry
 {
-	char id[24];
+	char id[24]	__attribute__((nonstring));
```

Comment:
While it is indeed compared with `memcmp()`, I wonder if it is OK to initialize 24-byte char array with 25-byte zero-terminated string.

Reply:
I decided not do do that based of the warning in the comment above the struct:
```
/*                                                                               
 * Database of block devices that lie about physical sector sizes.  The          
 * identification string must be precisely 24 characters to avoid false          
 * negatives                                                                     
 */ 
 ```

Reply:
SATA model strings are indeed not terminated.  I just wonder about a proper C syntax side, so that we won't stumble on this again on some later compiler update.

Reply:
The proper fix would be to change it to `char id[25]`.  The 24-byte memcmp() should still work fine.  I don't have one of the drives in the table to actually test the change though.

I just chose `__attribute__((nonstring))` since it's the most minimal change needed to make the compiler happy.

Reply:
I spoke too soon, I'm seeing failures in FreeBSD with `__attribute__((nonstring))`...
```
  /usr/home/zfs/zfs/module/zcommon/zfs_valstr.c:42:40: error: unknown attribute 'nonstring' ignored [-Werror,-Wunknown-attributes]
     42 |         const char      vb_pair[2]       __attribute__((nonstring));
        |                                                         ^~~~~~~~~
```